junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -286,15 +287,22 @@ static class CloseablePath implements CloseableResource {
 
 		private CloseablePath(TempDirFactory factory, CleanupMode cleanupMode, AnnotatedElementContext elementContext,
 				ExtensionContext extensionContext) throws Exception {
-			this.dir = validateTempDirectory(factory.createTempDirectory(elementContext, extensionContext));
+			this.dir = factory.createTempDirectory(elementContext, extensionContext);
 			this.factory = factory;
 			this.cleanupMode = cleanupMode;
 			this.extensionContext = extensionContext;
+
+			try {
+				validateTempDirectory(this.dir);
+			}
+			catch (PreconditionViolationException e) {
+				close();
+				throw e;
+			}
```

Comment:
Wouldn't this be simpler?

```suggestion
			if (dir == null || !Files.isDirectory(dir)) {
				close();
				throw new PreconditionViolationException("temp directory must be a directory");
			}
```

Reply:
Considering this change only, yes 🙂

I had in mind to add the file system check (#3910) to the `validateTempDirectory` body, i.e., the `CloseablePath` constructor validation from https://github.com/junit-team/junit5/issues/3910#issuecomment-2267594998.

For the sake of clarity, let me apply your suggestion here and I can reshape that in the PR for #3910.

Reply:
I see now how sticking to `Preconditions.condition` makes things more complicated. No matter how this would evolve, it's best to throw the `PreconditionViolationException` explicitly. Thanks for the hint!